lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -455,14 +455,24 @@ def do_3d_projection(self, renderer):
 
         fcs = (zalpha(self._facecolor3d, vzs) if self._depthshade else
                self._facecolor3d)
-        fcs = mcolors.to_rgba_array(fcs, self._alpha)
-        self.set_facecolors(fcs)
-
         ecs = (zalpha(self._edgecolor3d, vzs) if self._depthshade else
                self._edgecolor3d)
+
+        # Sort the points based on z coordinates
+        # Put vzs first to sort based on z value
+        z_markers = sorted(zip(vzs, np.column_stack([vxs, vys]), fcs, ecs),
+                           reverse=True)
+        vps = [ps for zs, ps, fc, ec in z_markers]
```

Comment:
Can you just do:
```python
[zzs, vps, fcs, ecs] = zip(*zmarkers)
```

Reply:
I knew there should've been an easier way to do that. I did not know about the starred expression syntax though. I have also changed up Poly3DCollection's do_3d_projection method to use this.